src/radical/pilot/agent/scheduler/continuous.py

```diff
@@ -70,258 +76,387 @@ def finalize_child(self):
+        default, we only allow for partial allocations on the first and last
+        node - but all intermediate nodes MUST be completely used.  This way we
+        allocate a *continuous* set of slots.
+
+        If the scheduler is configured with `scattered=True`, then that
+        constraint is relaxed, and any set of slots (continuous across nodes or
+        not) is accepted as valid allocation.
+
+        No matter the mode, we always make sure that we allocate in chunks of
+        'threads_per_proc', as otherwise the application would not be able to
+        spawn the requested number of threads on its slots.
+        '''
+
+        requested_procs  = cud['cpu_processes']
+        threads_per_proc = cud['cpu_threads']
+        requested_cores  = requested_procs * threads_per_proc
+        requested_gpus   = cud['gpu_processes']
+
+        # First and last nodes can be a partial allocation - all other nodes can
+        # only be partial when `scattered` is set.
+        #
+        # Iterate over all nodes until we find something. Check if it fits the
+        # allocation mode and sequence.  If not, start over.  If it does, add
+        # and continue to next node.
+        #
+        # FIXME: we should use a persistent node index to start searching where
+        #        we last left off, instead of always starting from the beginning
+        #        of the node list (optimization).
+
+        # Things are complicated by chunking: we only accept chunks of
+        # 'threads_per_proc', as otherwise threads would need to be distributed
+        # over nodes, which is not possible for multi-system-image clusters this
+        # scheduler assumes.  Thus scheduling will *always* fail if:
+        #
+        #   - requested_cores >= 3 * cores_per_node
+        #   - cores_per_node % threads_per_proc != 0
+        #   - scattered == False
+        #
+        # but it can fail for less cores, too, if the partial first and last
+        # allocation are not favorable.  We thus raise an exception for
+        # requested_cores > cores_per_node on impossible full-node-chunking
 
-        # Convenience aliases
         cores_per_node = self._lrms_cores_per_node
-        all_slots = self.slots
-
-        # Glue all slot core lists together
-        all_slot_cores = [core for node in [node['cores'] for node in all_slots] for core in node]
-        # self._log.debug("all_slot_cores: %s", all_slot_cores)
-
-        # Find the start of the first available region
-        all_slots_first_core_offset = self._find_cores_cont(all_slot_cores, 
-                cores_requested, rpc.FREE)
-        self._log.debug("all_slots_first_core_offset: %s", all_slots_first_core_offset)
-        if all_slots_first_core_offset is None:
+        gpus_per_node  = self._lrms_gpus_per_node
+        
+        if  requested_cores  > cores_per_node   and \
+            cores_per_node   % threads_per_proc and \
```

Comment:
I do not think the mode of cores per node with threads should concern the scheduler. If there are resources that will fit the unit it should be fine.